cli/tauri-bundler/src/bundle/osx_bundle.rs

```diff
@@ -76,9 +80,207 @@ pub fn bundle_project(settings: &Settings) -> crate::Result<Vec<PathBuf>> {
 
   create_path_hook(&bundle_directory, settings).chain_err(|| "Failed to create _boot wrapper")?;
 
+  if let Some(identity) = settings.osx_signing_identity() {
+    sign(app_bundle_path.clone(), identity, &settings)?;
+    match notarize_auth_args() {
+      Ok(args) => {
+        notarize(app_bundle_path.clone(), args, settings)?;
+      }
+      Err(e) => {
+        common::print_info(format!("skipping app notarize, {}", e.to_string()).as_str())?;
+      }
+    }
+  }
+
   Ok(vec![app_bundle_path])
 }
 
+fn sign(app_bundle_path: PathBuf, identity: &str, settings: &Settings) -> crate::Result<()> {
+  common::print_info(format!(r#"signing app with identity "{}""#, identity).as_str())?;
+  let mut args = vec![
+    "--deep",
+    "--force",
+    "-s",
+    identity,
```

Comment:
```suggestion
    identity,
    "--options",
    "runtime"
```

This is required in order to notarize an app:

```
runtime  On macOS versions >= 10.14.0, opts signed processes into a hard-
              ened runtime environment which includes runtime code signing
              enforcement, library validation, hard, kill, and debugging
              restrictions.  These restrictions can be selectively relaxed via
              entitlements. Note: macOS versions older than 10.14.0 ignore the
              presence of this flag in the code signature.
```